src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -400,7 +477,7 @@ private bool TryStopPreviousTimer(GrainReference grainRef, string reminderName)
         private async Task DoResponsibilitySanityCheck(GrainReference grainRef, string debugInfo)
         {
             if (status != ReminderServiceStatus.Started)
-                await startedTask.Task;
+                await startedTask.Task.WithTimeout(InitialLoadMaxWaitTimeForUpdates);
```

Comment:
so the callers will be getting this timeout if they call while the service is trying to init.
Instead, it would be better to do like you do now ` await startedTask.Task.WithTimeout(InitialLoadMaxWaitTimeForUpdates)`
but then catch and rethrow "service is still trying to init".